### PR TITLE
Initial drop as previously released manually

### PR DESCRIPTION
**Pull request by martinsmith1968**

### CODE REVIEW

File: FI.API.SignTool/FI.API.SignTool.csproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>netcoreapp2.1;netcoreapp2.2</TargetFrameworks>
```
Comment (IvanAlekseev) [inline FI.API.SignTool/FI.API.SignTool.csproj]: [2.1 is LTS and 3.0 is latest]()https://dotnet.microsoft.com/platform/support/policy/dotnet-core
```suggestion
    <TargetFrameworks>netcoreapp2.1;netcoreapp3.0</TargetFrameworks>
```

File: FI.API.SignTool/Helpers/CertificateHelper.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            }
+            catch
+            {
+                verified = false;
+            }
+
+            return verified;
+        }
+
+        private static byte[] HashBody(string content)
```
Comment (IvanAlekseev) [inline FI.API.SignTool/Helpers/CertificateHelper.cs]: We should probably work on naming in this class. HashBody makes sense as a whole, but not in this class which is named CertificateHelper

File: FI.API.SignTool/Program.cs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                return result;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine();
+                Console.Error.WriteLine($"Error: {ex.Message}");
+
+                return 2;
+            }
+#if !NETCOREAPP
```
Comment (IvanAlekseev) [inline FI.API.SignTool/Program.cs]: But it is NETCOREAPP

File: FI.API.SignTool/SigningProviders/AzureKeyVaultSigningProvider.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            return result.AccessToken;
+        }
+
+        public byte[] SignHash(byte[] bytes)
+        {
+            var client = GetClient();
+
+            var signedBytes = SignAsync(client, bytes)
+                .GetAwaiter()
+                .GetResult();
```
Comment (IvanAlekseev) [inline FI.API.SignTool/SigningProviders/AzureKeyVaultSigningProvider.cs:51]: Any chance we can do without async->sync conversion?

File: FI.API.SignTool/SigningProviders/KeyVaultConnectionString.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        public KeyVaultConnectionString(string connectionString)
+        {
+            var dict = (connectionString ?? string.Empty).Split(";".ToCharArray(), StringSplitOptions.RemoveEmptyEntries)
+                .Select(x => x.Split("=".ToCharArray()))
+                .ToDictionary(x => x.First().ToLower(), x => string.Join("=", x.Skip(1)));
+
+            Url           = GetValue(dict, nameof(Url).ToLower());
+            KeyName       = GetValue(dict, nameof(KeyName).ToLower());
+            KeyVersion    = GetValue(dict, nameof(KeyVersion).ToLower()) ?? string.Empty;
+            Algorithm     = GetValue(dict, nameof(Algorithm).ToLower()) ?? "RS256";
```
Comment (IvanAlekseev) [inline FI.API.SignTool/SigningProviders/KeyVaultConnectionString.cs:29]: Do we support anything else?

File: zipup.cmd [hunk 6]
```diff
@@ -0,0 +1,15 @@
+@ECHO OFF
```
Comment (IvanAlekseev) [inline zipup.cmd:1]: Why do we need this file?

**Review by IvanAlekseev — CHANGES_REQUESTED** [review]

File: FI.API.SignTool/FI.API.SignTool.csproj (hunk 1 above)
Comment (martinsmith1968) [reply FI.API.SignTool/FI.API.SignTool.csproj]: Removed multi targeting

File: FI.API.SignTool/Helpers/CertificateHelper.cs (hunk 2 above)
Comment (martinsmith1968) [reply FI.API.SignTool/Helpers/CertificateHelper.cs]: Agreed.

File: FI.API.SignTool/SigningProviders/AzureKeyVaultSigningProvider.cs (hunk 4 above)
Comment (martinsmith1968) [reply FI.API.SignTool/SigningProviders/AzureKeyVaultSigningProvider.cs:51]: ISigningProvider as not an async interface.

File: zipup.cmd (hunk 6 above)
Comment (martinsmith1968) [reply zipup.cmd:1]: Was previously used to distribute so left it in

File: FI.API.SignTool/Program.cs (hunk 3 above)
Comment (martinsmith1968) [reply FI.API.SignTool/Program.cs]: Yep

**Review by IvanAlekseev — APPROVED** [review]

**Pull request merged**